File: baek_joon/py_src/step11/2798.py
n, m = map(int, input().split())
n_list = list(map(int, input().split()))
sum = 0

for i in range(n-2):
    for j in range(i+1, n-1):
        for k in range(j+1, n):
            if n_list[i] + n_list[j] + n_list[k] > m:
                continue
            else:
                sum = max(sum, n_list[i] + n_list[j] + n_list[k]) 

print(sum)
    

File: baekjoon/js/2798.js
const fs = require("fs");
const buf = fs.readFileSync(0, "utf-8").toString().trim();

const inputs = buf.split("\n").map((input) => input.split(" ").map(Number));
const [, m] = inputs[0];
const nums = inputs[1];

let result = 0;

const findNum = (idx, sum, count) => {
  if (count === 3) {
    if (sum <= m) {
      result = Math.max(result, sum);
    }

    return;
  }

  for (let i = idx; i < nums.length; i++) {
    findNum(i + 1, sum + nums[i], count + 1);
  }
};

findNum(0, 0, 0);

console.log(result);
